aten/src/ATen/cuda/CUDAGraph.h

```diff
@@ -56,6 +82,15 @@ struct TORCH_CUDA_CPP_API CUDAGraph {
   // used to identify when a stream is participating in capture
   CaptureId_t capture_id_ = -1;
 
+  // How many dynamic input tensors does this graph expect?
```

Comment:
Outputs are much trickier than inputs because we don't control where/when they are allocated. The function can allocate memory wherever it wants, then, return it however it wants. Also, the CUDAGraph.cpp code (currently) has no way of knowing that information. On the Python side, you'd need to capture the return (arguably it should support returning multiple tensors) with all information (dtype, stride, offset, data_ptr) and feed it into CUDAGraph.cpp, which would work backwards through its allocations. Since allocations can overlap, it may need to become larger? e.g. `[0, 100]` and `[50, 150]` were allocations that we recorded, then the returned tensor is like `[70, 80]`. We don't actually know which of those the user code actually intended to return, since we're blindly poking through their allocations and guessing. So we'd have to allocate a 150-byte tensor and return [70:80] of it. A bit wasteful but I think it would work.